src/PaletteProvider.js

```diff
@@ -93,11 +100,48 @@ export default class MaterialPaletteProvider
     };
   }
 
+  _mergeWithDefaults(palette: PaletteInstance) {
+    const defaults = {
+      ...defaultSwatches,
+      ...Object.keys(this.props.defaults || {}).reduce(
+        (acc: *, profile: string) => ({
+          ...acc,
+          [profile]: { ...(this.props.defaults[profile] || {}), population: 0 },
+        }),
+        {},
+      ),
+    };
+    return {
+      ...Object.keys(palette)
+        .filter((profile: ColorProfile) => !!palette[profile]) // Stripping out unavailable profiles
+        .reduce(
+          (acc: *, profile: ColorProfile) => ({
+            ...acc,
+            [profile]: palette[profile],
+          }),
+          {},
+        ),
+      ...defaults,
+    };
+  }
+
+  _validateDefaults() {
+    if (this.props.defaults) {
+      if (typeof this.props.defaults !== 'object') {
```

Comment:
That part is not finished, I am about to complete the `defaults` validation today